src/main/ingestion/kafka-queue.ts

```diff
@@ -1,111 +1,81 @@
-            uuidOffset.set(event.uuid, message.offset)
-            return sanitizeEvent({
-                ...event,
-                site_url: event.site_url || null,
-                ip: event.ip || null,
-            })
+    private async eachMessage(message: KafkaMessage): Promise<void> {
+        const { data: dataStr, ...rawEvent } = JSON.parse(message.value!.toString())
+        const combinedEvent = { ...rawEvent, ...JSON.parse(dataStr) }
+        const event: PluginEvent = sanitizeEvent({
+            ...combinedEvent,
+            site_url: combinedEvent.site_url || null,
+            ip: combinedEvent.ip || null,
         })
+        await this.eachEvent(event)
+    }
 
-        const maxBatchSize = Math.max(
-            15,
-            Math.min(
-                100,
-                Math.ceil(
-                    pluginEvents.length / this.pluginsServer.WORKER_CONCURRENCY / this.pluginsServer.TASKS_PER_WORKER
-                )
-            )
-        )
+    private async eachEvent(event: PluginEvent): Promise<void> {
+        const eachEventStartTimer = new Date()
 
-        const processingTimeout = timeoutGuard('Still running plugins on events. Timeout warning after 30 sec!', {
-            eventCount: pluginEvents.length,
-            piscina: JSON.stringify(getPiscinaStats(this.piscina)),
+        const processingTimeout = timeoutGuard('Still running plugins on event. Timeout warning after 30 sec!', {
+            event: JSON.stringify(event),
         })
-        const processingBatches = groupIntoBatches(pluginEvents, maxBatchSize)
-        const processedEvents = (
-            await Promise.all(
-                processingBatches.map(async (batch) => {
-                    const timer = new Date()
-                    const processedBatch = this.processEventBatch(batch)
-                    this.pluginsServer.statsd?.timing('kafka_queue.single_event_batch', timer)
-                    return processedBatch
-                })
-            )
-        ).flat()
-
-        clearTimeout(processingTimeout)
-
-        this.pluginsServer.statsd?.timing('kafka_queue.each_batch.process_events', batchStartTimer)
-        const batchIngestionTimer = new Date()
-
-        // Sort in the original order that the events came in, putting any randomly added events to the end.
-        // This is so we would resolve the correct kafka offsets in order.
-        processedEvents.sort(
-            (a, b) => (uuidOrder.get(a.uuid!) || pluginEvents.length) - (uuidOrder.get(b.uuid!) || pluginEvents.length)
-        )
+        const timer = new Date()
+        let processedEvent: PluginEvent
+        try {
+            processedEvent = await this.processEvent(event)
+        } catch (error) {
+            status.info('🔔', error)
+            Sentry.captureException(error)
+            throw error
+        } finally {
+            this.pluginsServer.statsd?.timing('kafka_queue.single_event_batch', timer)
```

Comment:
This doesn't refer to a batch anymore, right?

Reply:
Nope. I just kept the name to have continuity in the graphs...

Reply:
I'd rename to something different though, will be confusing for people without the history of this

Reply:
ok, why not. removed `_batch`